internal/transport/server.go

```diff
@@ -76,6 +76,7 @@ func (s *Server) Stop(ctx context.Context) error {
 func (s *Server) initMiddleware() {
 	m := middleware{log: s.Logger, writer: s.writer, traceExtractor: s.TraceExtractor}
 	s.router.Use(m.recoverer)
+	s.router.Use(m.timeout(5 * time.Second))
```

Comment:
why is this needed?

Reply:
To limit the max time a request can be at the server. I'm seeing huge latency spikes at the metrics, this will make sure those spikes will reach 5 seconds at the maximum and help the user because after 5 seconds this is likely to fail anyway.